services/community-site/src/utils/validators.ts

```diff
@@ -0,0 +1,34 @@
+import { Validator, ValidatorWithStats, YieldedValidator } from '../interfaces/Validator';
+
+const calulcateValidatorYield = (validators: Validator[]): ValidatorWithStats[] | Validator[] => {
+  const hasStats = (validators[0] as ValidatorWithStats)?.pbftsProduced !== 0;
+  if (!hasStats) {
+    return validators;
+  }
+  const validatorsWithStats = validators as ValidatorWithStats[];
+  const validatorsWithStake: YieldedValidator[] = validatorsWithStats.map((v) => {
+    return {
+      ...v,
+      blocksPerStake: v.pbftsProduced || 0 / Number.parseFloat(v.delegation.toString()),
+    };
+  });
+  const minBlockRatio = validatorsWithStake.reduce((prev, curr) =>
+    prev.blocksPerStake! < curr.blocksPerStake! ? prev : curr,
+  );
+
+  const maxBlockRatio = validatorsWithStake.reduce((prev, curr) =>
+    prev.blocksPerStake! > curr.blocksPerStake! ? prev : curr,
+  );
+  return validatorsWithStake.map((validator) => {
+    const yieldRatio =
+      ((parseFloat(validator.blocksPerStake.toString()) - minBlockRatio.blocksPerStake) /
```

Comment:
Si previously `validator.blocksPerStake` was a number 'cause you were comparing it and now it's not?

Reply:
float parsing requires a string so we need to convert it before